includes/class-wc-subscriptions-synchroniser.php

```diff
@@ -687,7 +687,7 @@ public static function calculate_first_payment_date( $product, $type = 'mysql',
 				}
 			}
 			// when a certain number of months are added and the first payment date moves to next year
-			if ( $month_number < gmdate( 'm', $from_timestamp ) ) {
+			if ( $month_number <= gmdate( 'm', $from_timestamp ) ) {
```

Comment:
I wonder if we need to move this whole if-else block. This whole section seems far more complicated than it needs to be, however it seems like in the if-block above this one the logic is determining if the customer is purchasing before or after the payment day. 

I might be wrong, however I think we'd only ever need to add a year if the customer is purchasing after the day. ie we're adding a month.

Reply:
wait no, that wouldn't be possible in cases when we have a product with x months as the interval. In the issue the product has a 12 month interval and in cases like that we do add x number of months in both cases that could mean we need to add a year.